### PR TITLE
Avoid harness error in applyContraints/getConstraints test

### PR DESCRIPTION
**Pull request by foolip**
Because the promise callbacks containing the asserts weren't wrapped
in `t.step_func` a failed assert turned into a harness error:
https://staging.wpt.fyi/results/mediacapture-image/MediaStreamTrack-getConstraints-fast.html?run_id=227500002&run_id=259130004&run_id=253190002&run_id=247230003

This masked that the tests were actually failing.

### CODE REVIEW

File: mediacapture-image/MediaStreamTrack-getConstraints-fast.html [hunk 1]
```diff
@@ -35,45 +35,31 @@
...
-    // service in this Layout Test, but |constraintsIn| should be cached.
-    videoTrack.applyConstraints(constraintsIn)
-        .then(() => { /* ignore */ })
-        .catch((e) => { /* ignore */ })
-        .then(() => {
-          const constraintsOut = videoTrack.getConstraints();
-          assert_object_equals(constraintsOut, constraintsIn, "constraints");
-          t.done();
-        });
+    await videoTrack.applyConstraints(constraintsIn);
```
Comment (foolip) [inline mediacapture-image/MediaStreamTrack-getConstraints-fast.html]: Note that this changes the behavior of the test. This being rejected is no longer tolerated. I tried wrapping this in try/catch, but even if one suppresses the error the following assert still fails. Feedback appreciated.

**Comment by foolip** [comment]
https://wpt.fyi/results/mediacapture-image/MediaStreamTrack-getConstraints-fast.html?run_id=313930003&run_id=303090005&run_id=277420005 shows that this will cause all tests to fail in all browsers instead.

Reviewers, input much appreciated on what this test *should* do.

Context: consistent harness error usually indicate a problem with the test, and there aren't very many of them. It would be nice if there were none and we could add tooling to prevent them from being introduced.

File: mediacapture-image/MediaStreamTrack-getConstraints-fast.html (hunk 1 above)
Comment (reillyeon) [reply mediacapture-image/MediaStreamTrack-getConstraints-fast.html]: My recollection is that `async_test` runs in parallel while `promise_test` runs in sequence. The failure may be caused by the original test being buggy. Can you upload the error where you wrap the `applyConstraints()` call in a try/catch to ignore any errors?

**Review by reillyeon — CHANGES_REQUESTED** [review]

File: mediacapture-image/MediaStreamTrack-getConstraints-fast.html (hunk 1 above)
Comment (foolip) [reply mediacapture-image/MediaStreamTrack-getConstraints-fast.html]: That's right, `promise_test` run in sequence. The way this was written it should have been fine to race all the subtests with `async_test`, I switched to `promise_test` just to make the test clearer.

Comment (foolip) [reply mediacapture-image/MediaStreamTrack-getConstraints-fast.html]: I've pushed a change with the try/catch to show what happens.

Comment (reillyeon) [reply mediacapture-image/MediaStreamTrack-getConstraints-fast.html]: I'm sorry but I'm having some trouble navigating the CI results. What is the error causing the failure? If it is an `assert_object_equals()` what are the actual and expected values?

**Comment by foolip** [comment]
Looks like Taskcluster was failing due to https://github.com/web-platform-tests/wpt/pull/18930 (fixed in https://github.com/web-platform-tests/wpt/pull/18962) so results for Chrome and Firefox weren't produced. I'll rebase this branch to retrigger.

File: mediacapture-image/MediaStreamTrack-getConstraints-fast.html (hunk 1 above)
Comment (foolip) [reply mediacapture-image/MediaStreamTrack-getConstraints-fast.html]: After we were embarrassed at TPAC to realize how obscure the "Expand" button is and you learned of its existence, can you tell from the results at which step this failed and if you think this is the right failure mode?

**Comment by foolip** [comment]
@reillyeon https://wpt.fyi/results/?pr=18931&label=pr_head&max-count=1 is the view to look at. The failure still isn't very obvious, but what's happening is that the "advanced" property is missing from the return value of `getContraints()`.

**Pull request closed without merging**